### ISSUE TITLE
Maintain freshness on the result of onbject spread operator

### ISSUE THREAD

**Issue by mhegazy** [Microsoft/TypeScript#12717]
```ts
interface I { 
    a: number;
    b: string;
}

var obj = { c: true };

var i: I = { a: 0, b: "s", c: obj.c };// Error, unknown property c
var i: I = { a: 0, b: "s", ...obj } // No error here.
```

**Comment by mhegazy** [comment]
As noted in https://github.com/Microsoft/TypeScript/issues/2103#issuecomment-265534846, spreading a symbolic name is different from adding the property definition in the object literal. in the later, an excess property is likely an error; in the earlier it is not necessarily one.

**Comment by aaronbeall** [comment]
@mhegazy 

Why was this closed? Maybe I didn't understand but this is still an issue:

```
interface Foo {
  a: number;
  b: string;
}

declare const stuff: any;

const a: Foo = { a: 1, b: "b" } // Ok
const b: Foo = { a: 1, b: "b", c: "c" } // Error - Ok
const c: Foo = { a: 1, b: "b", c: "c", ...stuff } // No Error - not Ok!
```

**Comment by voidale** [comment]
I'm facing the same error why was that declined?

**Comment by mhegazy** [comment]
An update on this issue. we have revisited the original decision (more in https://github.com/Microsoft/TypeScript/issues/14853), and these should be flagged now as errors in TS 2.4, see https://github.com/Microsoft/TypeScript/issues/15908.

**Comment by aaronbeall** [comment]
@mhegazy That's great news! Thanks for the update!
